packages/oas-to-snippet/src/languages.ts

```diff
@@ -0,0 +1,367 @@
+
+  Object.entries(targets).forEach(([target, { clientsById }]) => {
+    if (!(target in languages)) {
+      return;
+    }
+
+    Object.entries(clientsById).forEach(([client, data]) => {
+      if (!(client in languages[target as SupportedTargets].httpsnippet.targets)) {
+        return;
+      }
+
+      if ('installation' in data.info) {
+        languages[target as SupportedTargets].httpsnippet.targets[client].install = data.info.installation;
+      }
+    });
+  });
+
+  plugins.forEach(({ target, client: { info: clientInfo } }) => {
+    const clientKey = clientInfo.key;
+
+    languages[target as SupportedTargets].httpsnippet.targets[clientKey] = {
+      name: clientInfo.title,
+    };
+
+    if (clientInfo.installation) {
+      languages[target as SupportedTargets].httpsnippet.targets[clientKey].install = clientInfo.installation;
+    }
+  });
+
+  return languages;
+}
+
+export function getLanguageConfig(languages: SupportedLanguages, lang: Language) {
+  let config: LanguageConfig;
+  let language: TargetId;
+  let target: ClientId;
+
+  switch (lang) {
+    case 'curl':
+      config = languages.shell;
+      language = 'shell';
+      target = 'curl';
+      break;
+
+    // Our `api` snippet plugin, `httpsnippet-client-api`, that we use in ReadMe used to be called
+    // `node-simple`. For backwards compatibility we're retaining this old identifier here.
+    case 'node-simple':
+      config = languages.node;
+      language = 'node';
+      target = 'api';
+      break;
+
+    default:
+      // If `lang` is an array, then it's a mixture of language and targets like `[php, guzzle]` or
+      // `[javascript, axios]` so we need to a bit of work to pull out the necessary information
+      // needed to build the snippet.
+      if (Array.isArray(lang)) {
+        if (lang[0] in languages) {
+          if (lang[1] in languages[lang[0] as SupportedTargets].httpsnippet.targets) {
+            config = languages[lang[0] as SupportedTargets];
+            language = config.httpsnippet.lang;
+            target = lang[1];
+          }
+        }
+      } else if (lang in languages) {
+        config = languages[lang];
+        language = config.httpsnippet.lang;
+        target = config.httpsnippet.default;
+      } else {
+        throw new Error('An unknown language was supplied.');
+      }
+  }
+
+  return {
+    config,
+    language,
+    target,
+  };
+}
+
+export function getClientInstallationInstructions(
+  languages: SupportedLanguages,
+  lang: Language,
+  /**
+   * For `api` snippet generation we need supply a ReadMe API Registry identifier that'll be used
+   * to retrieve the OpenAPI definition to generate their SDK through the `npx api install` command.
+   * These users won't be installing the `api` package itself to run the snippets we generate for
+   * them.
+   *
+   * @example @developers/v2.0#17273l2glm9fq4l5
+   */
+  registryIdentifier?: string,
```

Comment:
i imagine other plugins might want custom installation steps similar to `api` does, could we make this (and its corresponding JSDocs) more generic (i.e., call it `packageName`)?

Reply:
this work is a little confusing but the `registryIdentifier` here is the `npx api install <registry identifier>` part of things and `variableName` is the eventual post-installation package name.

Reply:
could maybe have the config for the plugin supplied alongside the plugin but that's going to require a **lot** of work to httpsnippet as it doesn't support options like this

```js
const snippet = await oasToSnippet(
  petstore,
  petstore.operation('/user/login', 'get'),
  formData,
  auth,
  // `[node, api]` is not a standard language in `oas-to-snippet` but is
  // dynamically made available via this loaded plugin.
  ['node', 'api'],
  {
    plugins: [httpsnippetClientAPIPlugin({ api: { registryIdentifier: '@petstore/v2.0#17273l2glm9fq4l5', }})],
  },
);
```